src/transformers/agents/agents.py

```diff
@@ -911,7 +1029,11 @@ def step(self):
         try:
             result = self.python_evaluator(
                 code_action,
-                tools=self.available_tools,
+                tools={
+                    **BASE_PYTHON_TOOLS.copy(),
+                    **self.toolbox.tools,
+                },
+                custom_tools=self.custom_tools,
```

Comment:
This change is made because I needed to separate two sets of tools:
- user-defined tools that are static (at least they do not change within one call to `run`) but can be changed in-between `run` calls
- agent-defined tools that will have to change within one run (thus the different handling in python interpreter)